pczt/src/roles/creator/mod.rs

```diff
@@ -0,0 +1,64 @@
+use std::collections::BTreeMap;
+
+use crate::{Pczt, V5_TX_VERSION, V5_VERSION_GROUP_ID};
+
+pub struct Creator {
```

Comment:
Could we link to the struct it's building in the struct docs?

Nitpick: Could we rename this to `PcztBuilder` so it's more immediately obvious that it's following the builder pattern and that it's for the other struct?

Reply:
This is not a PCZT builder; this is a struct implementing the Creator role, specifically as intended to follow the (applicable) roles introduced in BIP 174 and BIP 370 for PSBTs.

The "PCZT builder" is in effect spread across the Creator and Constructor roles, and arguably the Updater role as well.